helmchart/sdp/values.yaml

```diff
@@ -4,116 +4,178 @@
+  ## @param sdp.route.domain Public domain/address of the SDP service. If using localhost, consider including the port as part of the domain.
+  ## @param sdp.route.port Primary port on which the SDP service listens.
+  ## @param sdp.route.metricsPort Port dedicated to metrics collection for the SDP service.
   route:
     schema: "https"
-    # The domain is the public address of that service. For localhost you will want to include the port as part of the domain.
-    # domain: "localhost:8000"
     domain: sdp.localhost.com
     port: "8000"
     metricsPort: "8002"
-  # =========================== START sdp.image ===========================
+
+  ## @extra sdp.image Configuration related to the Docker image used by the SDP service.
+  ## @param sdp.image.repository Docker image repository for the SDP backend service.
+  ## @param sdp.image.pullPolicy Image pull policy for the SDP service. For locally built images, consider using "Never" or "IfNotPresent".
+  ## @param sdp.image.tag Docker image tag for the SDP service. If set, this overrides the default value from `.Chart.AppVersion`.
   image:
-    # replace the `repository` with the actual image name.
     repository: stellar/stellar-disbursement-platform-backend
-    # for locally built images, use `pullPolicy: Never or IfNotPresent`.
     pullPolicy: IfNotPresent
-    # If `tag` is set, it'll override the default value set in `.Chart.AppVersion`.
     tag: "latest"
 
-  # =========================== START sdp.deployment ===========================
+  ## @extra sdp.deployment Configuration related to the deployment of the SDP service.
+  ## @param sdp.deployment.annotations Annotations to be added to the deployment.
+  ## @extra sdp.deployment.podAnnotations Annotations specific to the pods, mainly for integration with monitoring tools like Prometheus.
+  ## @param sdp.deployment.podAnnotations.prometheus.io/path The path where Prometheus metrics are exposed.
+  ## @param sdp.deployment.podAnnotations.prometheus.io/port The port on which Prometheus metrics are exposed. This should match the SDP metrics port.
+  ## @param sdp.deployment.podAnnotations.prometheus.io/scrape Indicates if Prometheus should scrape metrics from this pod.
+  ## @param sdp.deployment.podSecurityContext Security settings for the pods.
+  ## @param sdp.deployment.securityContext Security settings for the container within the pod.
+  ## @extra sdp.deployment.strategy Configuration related to the deployment strategy, ensuring smooth updates and minimal downtime.
+  ## @param sdp.deployment.strategy.type Specifies the deployment strategy, RollingUpdate is preferred for minimal downtime.
+  ## @param sdp.deployment.strategy.rollingUpdate.maxUnavailable The maximum number of pods that can be unavailable during an update.
+  ## @param sdp.deployment.strategy.rollingUpdate.maxSurge The maximum number of pods that can be created over the desired number of pods.
```

Comment:
The same goes for the inner configuration under strategy, all the strategy section is applied to the templates as-is:
https://github.com/stellar/stellar-disbursement-platform-backend/blob/2945cef15ca51f0fbba99b3c45cbdbb4845cd329/helmchart/sdp/templates/02-deployment-ap.yaml#L20-L23

The same applies for all:
- [ ] sdp.deployment.strategy
- [ ] anchorPlatform.deployment.strategy
- [ ] tss.deployment.strategy
- [ ] dashboard.deployment.strategy